From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.j.ledkov@intel.com>
Date: Tue, 17 Mar 2015 14:59:12 +0000
Subject: [PATCH] Do not reuse mock group, as it might be defined in the nested
 chroot.

Rebased for mock 2.4 update by Francisco Boni <boboniboni@gmail.com>.
Rebased for mock 2.12 update by Patrick McCarty.
Rebased for mock 3.0 update by Patrick McCarty.

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 mock/py/mockbuild/config.py | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/mock/py/mockbuild/config.py b/mock/py/mockbuild/config.py
index 9a16a96..58d217a 100644
--- a/mock/py/mockbuild/config.py
+++ b/mock/py/mockbuild/config.py
@@ -49,7 +49,7 @@ def nspawn_supported():
 
 
 @traceLog()
-def setup_default_config_opts(unprivUid):
+def setup_default_config_opts(unprivUid, unprivGid=None):
     "sets up default configuration."
     config_opts = TemplatedDictionary(alias_spec={'dnf.conf': ['yum.conf']})
     config_opts['config_paths'] = []
@@ -65,11 +65,14 @@ def setup_default_config_opts(unprivUid):
     config_opts['log_config_file'] = 'logging.ini'
     config_opts['rpmbuild_timeout'] = 0
     config_opts['chrootuid'] = unprivUid
-    try:
-        config_opts['chrootgid'] = grp.getgrnam("mock")[2]
-    except KeyError:
-        #  'mock' group doesn't exist, must set in config file
-        pass
+    if unprivGid:
+        config_opts['chrootgid'] = unprivGid
+    else:
+        try:
+            config_opts['chrootgid'] = grp.getgrnam("mock")[2]
+        except KeyError:
+            #  'mock' group doesn't exist, must set in config file
+            pass
     config_opts['chrootgroup'] = 'mock'
     config_opts['chrootuser'] = 'mockbuild'
     config_opts['build_log_fmt_name'] = "unadorned"
@@ -726,10 +729,12 @@ def do_update_config(log, config_opts, cfg, uidManager, name, skipError=True):
 @traceLog()
 def load_defaults(uidManager):
     if uidManager:
-        gid = uidManager.unprivUid
+        uid = uidManager.unprivUid
+        gid = uidManager.unprivGid
     else:
-        gid = os.getuid()
-    return setup_default_config_opts(gid)
+        uid = os.getuid()
+        gid = os.getgid()
+    return setup_default_config_opts(uid, unprivGid=gid)
 
 
 def print_description(config_path, config_filename, uidManager):
